pkg/reconciler/pipelinerun/resources/validate_params_test.go

```diff
@@ -121,3 +121,83 @@ func TestValidateParamTypesMatching_Invalid(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateExpectedParametersProvided_Valid(t *testing.T) {
+	tcs := []struct {
+		name          string
+		p             *v1beta1.Pipeline
+		pr            *v1beta1.PipelineRun
+		errorExpected bool
+	}{{
+		name: "expected params provided",
+		p: tb.Pipeline("a-pipeline", tb.PipelineSpec(
+			tb.PipelineParamSpec("expected-string-param", v1beta1.ParamTypeString),
+			tb.PipelineParamSpec("expected-array-param", v1beta1.ParamTypeArray))),
+		pr: tb.PipelineRun("a-pipelinerun", tb.PipelineRunSpec("test-pipeline",
+			tb.PipelineRunParam("expected-string-param", "somestring"),
+			tb.PipelineRunParam("expected-array-param", "another", "array"))),
+		errorExpected: false,
+	}, {
+		name:          "no params to get wrong",
+		p:             tb.Pipeline("a-pipeline"),
+		pr:            tb.PipelineRun("a-pipelinerun"),
+		errorExpected: false,
+	}, {
+		name: "expected string param missing",
+		p: tb.Pipeline("a-pipeline", tb.PipelineSpec(
+			tb.PipelineParamSpec("expected-string-param", v1beta1.ParamTypeString),
+			tb.PipelineParamSpec("expected-array-param", v1beta1.ParamTypeArray))),
+		pr: tb.PipelineRun("a-pipelinerun", tb.PipelineRunSpec("test-pipeline",
+			tb.PipelineRunParam("expected-array-param", "another", "array"))),
+		errorExpected: true,
+	}, {
+		name: "expected array param missing",
+		p: tb.Pipeline("a-pipeline", tb.PipelineSpec(
+			tb.PipelineParamSpec("expected-string-param", v1beta1.ParamTypeString),
+			tb.PipelineParamSpec("expected-array-param", v1beta1.ParamTypeArray))),
+		pr: tb.PipelineRun("a-pipelinerun", tb.PipelineRunSpec("test-pipeline",
+			tb.PipelineRunParam("expected-string-param", "somestring"))),
+		errorExpected: true,
+	}}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			err := ValidateExpectedParametersProvided(&tc.p.Spec, tc.pr)
+			if (!tc.errorExpected) && (err != nil) {
+				t.Errorf("Pipeline.Validate() returned error: %v", err)
```

Comment:
i think it's helpful to add a bit more context here so if this fails the person running the tests has more info about whats going on, e.g. something like "didnt expect valid pipelinerun to return error but got %v"

Reply:
your error in the next test is much more descriptive!